common/app/com/commercetools/sunrise/search/pagination/viewmodels/AbstractPaginationViewModelFactory.java

```diff
@@ -121,11 +125,12 @@ private long calculateTopThresholdPage(final long currentPage, final long totalP
         return Math.min(currentPage + paginationSettings.getDisplayedPages(), totalPages);
     }
 
-    private long calculateTotalPages(final PagedResult<?> pagedResult, final long currentPage) {
-        if (pagedResult.isLast()) {
-            return currentPage;
-        } else if (pagedResult.getCount() > 0) {
-            final Double totalPages = Math.ceil(pagedResult.getTotal() / pagedResult.getCount());
+    private long calculateTotalPages(final PagedResult<?> pagedResult) {
+        final Long count = pagedResult.getCount();
+        if (count > 0) {
+            final Long total = pagedResult.getTotal();
+            final Http.Context context = Http.Context.current();
+            final Double totalPages = Math.ceil(((double) total) / entriesPerPageFormSettings.getLimit(context));
```

Comment:
This class had some bugs which I found when i added the pagination for the wishlist. But since we decided to remove the pagination for the wishlist, I can also revert these changes.